proto/list.proto

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright (c) 2024 Diagrid Inc.
+Licensed under the MIT License.
+*/
+
+syntax = "proto3";
+
+package api;
+
+import "proto/job.proto";
+
+option go_package = "github.com/diagridio/go-etcd-cron/api";
+
+message NamedJob {
+  // name is the name of the job.
+  string name = 1;
+
+  // job is the scheduled job.
+  Job job = 2;
+}
+
+// ListResponse is returned by the List RPC.
+message ListResponse {
+  // jobs is the list of jobs.
+  repeated NamedJob jobs = 1;
+
+  // more indicates there are more jobs in the given prefix which do not fit in
+  // the response size.
+	bool more = 2;
```

Comment:
nit spacing

Reply:
this handles pagination?

Reply:
Thanks :+1: 

No, we don't handle any kind of pagination or filtering etc. We simply request the most amount we can from Etcd. I updated the [API call](https://github.com/diagridio/go-etcd-cron/pull/43/files#diff-5b9b550fff634f1fd5596893d63ceeb6afa6a355427db919a644a21f097b7834R190) to ensure we get the upper limit using a [parameter of `0`](https://pkg.go.dev/go.etcd.io/etcd/client/v3#WithLimit). Indeed, the client is out of luck if their request is larger than the limit for now. We can add further parameters on the `List` request to resolve this issue in future, though I thought that at least adding `more`, the client has a chance of understand what's occurring.